src/ess/amor/geometry.py

```diff
@@ -0,0 +1,64 @@
+import scipp as sc
+
+
+class Detector:
+    'Description of the geometry of the Amor detector'
+    # number of active blades in the detector
+    nBlades = sc.scalar(14)
+    # number of wires per blade
+    nWires = sc.scalar(32)
+    # number of stipes per blade
+    nStripes = sc.scalar(64)
+    # angle of incidence of the beam on the blades (def: 5.1)
+    angle = sc.scalar(5.1, unit='degree').to(unit='rad')
+    # height-distance of neighboring pixels on one blade
+    dZ = sc.scalar(4.0, unit='mm') * sc.sin(angle)
+    # depth-distance of neighboring pixels on one blade
+    dX = sc.scalar(4.0, unit='mm') * sc.cos(angle)
+    # distance between detector blades
+    bladeZ = sc.scalar(10.455, unit='mm')
+    # vertical center of the detector
+    zero = 0.5 * nBlades.value * bladeZ
+    # distance from focal point to leading blade edge
+    distance = sc.scalar(4000, unit='mm')
+
+
+def pixel_coordinate_in_detector_system(pixelID):
+    """determine spatial coordinates and angles from pixel number"""
+    pixelID.unit = ''
```

Comment:
Still need to modify the input here to work around the unit issues.
We want the pixel id:s to have no unit, but the id:s are used to construct the geometry, so they need to be treated as dimensionless numbers in this context.

But have added a type hint and fixed the input modification issue.

Reply:
I worked around this by computing a `pixelID` before calling the function with a dimensionless unit.
I didn't like the changing unit and changing back at the end, because if an error occurs in the middle of the function, we have modified the input.